pkg/controller/broker/v2alpha2/activemqartemis/activemqartemis_reconciler.go

```diff
@@ -242,47 +248,65 @@ func sourceEnvVarFromSecret(customResource *brokerv2alpha2.ActiveMQArtemis, curr
 		Namespace: currentStatefulSet.Namespace,
 	}
 	// Attempt to retrieve the secret
-	stringDataMap := map[string]string{
-		envVarName: acceptorEntry,
+	stringDataMap := make(map[string]string)
+
+	for k := range *envVars {
+		stringDataMap[k] = (*envVars)[k]
 	}
+
 	nettySecret := secrets.NewSecret(namespacedName, secretName, stringDataMap)
 	if err = resources.Retrieve(namespacedName, client, nettySecret); err != nil {
 		if errors.IsNotFound(err) {
 			requestedResources = append(requestedResources, nettySecret)
-			//err = resources.Create(customResource, client, scheme, nettySecret)
 		}
 	} else { // err == nil so it already exists
 		// Exists now
 		// Check the contents against what we just got above
-		elem, ok := nettySecret.Data[envVarName]
-		if 0 != strings.Compare(string(elem), acceptorEntry) || !ok {
-			nettySecret.Data[envVarName] = []byte(acceptorEntry)
 
+		var needUpdate bool = false
+		for k := range *envVars {
+			elem, ok := nettySecret.Data[k]
+			if 0 != strings.Compare(string(elem), (*envVars)[k]) || !ok {
+				log.Info("Secret exists but not equals, or not ok", "elem", string(elem), "entry", (*envVars)[k], "ok?", ok)
+				nettySecret.Data[k] = []byte((*envVars)[k])
+				needUpdate = true
+			}
+		}
+
+		if needUpdate {
 			// These updates alone do not trigger a rolling update due to env var update as it's from a secret
 			err = resources.Update(namespacedName, client, nettySecret)
 
 			// Force the rolling update to occur
 			environments.IncrementTriggeredRollCount(currentStatefulSet.Spec.Template.Spec.Containers)
+
+			//so far it doesn't matter what the value is as long as it's greater than zero
 			retVal = statefulSetAcceptorsUpdated
 		}
 	}
 
-	acceptorsEnvVarSource := &corev1.EnvVarSource{
-		SecretKeyRef: &corev1.SecretKeySelector{
-			LocalObjectReference: corev1.LocalObjectReference{
-				Name: secretName,
+	log.Info("Pupulating env vars", "envVars", *envVars)
```

Comment:
In updating this line to be
log.V(5).Info("Populating...)
to address it logging passwords in the clear from the env vars I had anticipated a quick test that would show it only now logged in debug logging configuration.

However instead I observed a problem where my log file was growing exponentially with entries like:
`
{"level":"info","ts":1591010195.7600157,"logger":"controller_v2alpha2activemqartemis","msg":"Environment is OpenShift"}
{"level":"info","ts":1591010195.760062,"logger":"controller_v2alpha2activemqartemis","msg":"Checking routeDefinition for wconsj-2960"}
{"level":"info","ts":1591010195.7601032,"logger":"package k8s_actions","msg":"Configuring *v1.Route","ActiveMQArtemis Name ":"ex-aao-wconsj-2960-svc-rte"}
{"level":"info","ts":1591010195.7601504,"logger":"package k8s_actions","msg":"Retrieving *v1.Route","ActiveMQArtemis Name":"ex-aao-wconsj-2960-svc-rte"}
{"level":"info","ts":1591010195.7602043,"logger":"package k8s_actions","msg":"*v1.Route IsNotFound","ActiveMQArtemis Name":"ex-aao-wconsj-2960-svc-rte","Namespace":"entmqbr-3021-0","Name":"ex-aao-wconsj-2960-svc-rte"}
{"level":"info","ts":1591010195.7602649,"logger":"package k8s_actions","msg":"ex-aao-wconsj-2960-svc-rte not found","ActiveMQArtemis Name ":"ex-aao-wconsj-2960-svc-rte"}
{"level":"info","ts":1591010195.7603328,"logger":"package k8s_actions","msg":"Configuring *v1.Service","ActiveMQArtemis Name ":"ex-aao-wconsj-2961-svc"}
{"level":"info","ts":1591010195.7603598,"logger":"package k8s_actions","msg":"Retrieving *v1.Service","ActiveMQArtemis Name":"ex-aao-wconsj-2961-svc"}
{"level":"info","ts":1591010195.7604024,"logger":"package k8s_actions","msg":"*v1.Service IsNotFound","ActiveMQArtemis Name":"ex-aao-wconsj-2961-svc","Namespace":"entmqbr-3021-0","Name":"ex-aao-wconsj-2961-svc"}
{"level":"info","ts":1591010195.7604246,"logger":"package k8s_actions","msg":"ex-aao-wconsj-2961-svc not found","ActiveMQArtemis Name ":"ex-aao-wconsj-2961-svc"}
{"level":"info","ts":1591010195.7604415,"logger":"package environments","msg":"Detect if openshift is running"}
{"level":"info","ts":1591010195.7659853,"logger":"controller_v2alpha2activemqartemis","msg":"Environment is OpenShift"}
{"level":"info","ts":1591010195.76603,"logger":"controller_v2alpha2activemqartemis","msg":"Checking routeDefinition for wconsj-2961"}
{"level":"info","ts":1591010195.7660694,"logger":"package k8s_actions","msg":"Configuring *v1.Route","ActiveMQArtemis Name ":"ex-aao-wconsj-2961-svc-rte"}
{"level":"info","ts":1591010195.7661114,"logger":"package k8s_actions","msg":"Retrieving *v1.Route","ActiveMQArtemis Name":"ex-aao-wconsj-2961-svc-rte"}
{"level":"info","ts":1591010195.7661986,"logger":"package k8s_actions","msg":"*v1.Route IsNotFound","ActiveMQArtemis Name":"ex-aao-wconsj-2961-svc-rte","Namespace":"entmqbr-3021-0","Name":"ex-aao-wconsj-2961-svc-rte"}
{"level":"info","ts":1591010195.7662568,"logger":"package k8s_actions","msg":"ex-aao-wconsj-2961-svc-rte not found","ActiveMQArtemis Name ":"ex-aao-wconsj-2961-svc-rte"}
{"level":"info","ts":1591010195.766297,"logger":"package k8s_actions","msg":"Configuring *v1.Service","ActiveMQArtemis Name ":"ex-aao-wconsj-2962-svc"}
{"level":"info","ts":1591010195.766323,"logger":"package k8s_actions","msg":"Retrieving *v1.Service","ActiveMQArtemis Name":"ex-aao-wconsj-2962-svc"}
{"level":"info","ts":1591010195.7663567,"logger":"package k8s_actions","msg":"*v1.Service IsNotFound","ActiveMQArtemis Name":"ex-aao-wconsj-2962-svc","Namespace":"entmqbr-3021-0","Name":"ex-aao-wconsj-2962-svc"}
{"level":"info","ts":1591010195.7663784,"logger":"package k8s_actions","msg":"ex-aao-wconsj-2962-svc not found","ActiveMQArtemis Name ":"ex-aao-wconsj-2962-svc"}
{"level":"info","ts":1591010195.7663949,"logger":"package environments","msg":"Detect if openshift is running"}
`
etc... until:
`
{"level":"info","ts":1591010205.444813,"logger":"controller_v2alpha2activemqartemis","msg":"Checking routeDefinition for wconsj-4999"}
{"level":"info","ts":1591010205.4449363,"logger":"package k8s_actions","msg":"ex-aao-wconsj-4999-svc-rte not found","ActiveMQArtemis Name ":"ex-aao-wconsj-4999-svc-rte"}
{"level":"info","ts":1591010205.4449663,"logger":"controller_v2alpha2activemqartemis","msg":"Entering into  Process Upgrade","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.4453986,"logger":"controller_v2alpha2activemqartemis","msg":"Reconcile loop did not find any resource changes to apply"}
{"level":"info","ts":1591010205.4454718,"logger":"controller_v2alpha2activemqartemis","msg":"Exiting CreatingK8sResourceState","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.4455006,"logger":"controller_v2alpha2activemqartemis","msg":"Entering ScalingState","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.44556,"logger":"controller_v2alpha2activemqartemis","msg":"Updating status for pods","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.4455862,"logger":"controller_v2alpha2activemqartemis","msg":"Getting status for pods","ActiveMQArtemis Name":"ex-aao-ss"}
{"level":"info","ts":1591010205.4456415,"logger":"olm","msg":"Found deployments with status ","stopped":[],"starting":["ex-aao-ss-1"],"ready":[]}
{"level":"info","ts":1591010205.4456666,"logger":"controller_v2alpha2activemqartemis","msg":"lastStatus.Ready len is \u0000"}
{"level":"info","ts":1591010205.4456847,"logger":"controller_v2alpha2activemqartemis","msg":"status.Ready len is \u0000"}
{"level":"info","ts":1591010205.462332,"logger":"controller_v2alpha2activemqartemis","msg":"Pods status updated","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.462437,"logger":"controller_v2alpha2activemqartemis","msg":"Reconciling ActiveMQArtemis","Request.Namespace":"entmqbr-3021-0","Request.Name":"ex-aao"}
{"level":"info","ts":1591010205.4624918,"logger":"controller_v2alpha2activemqartemis","msg":"Reconciling ActiveMQArtemis","Request.Namespace":"entmqbr-3021-0","Request.Name":"ex-aao","Operator version":"0.14.0-SNAPSHOT","ActiveMQArtemis release":"7.7.0"}
{"level":"info","ts":1591010205.4625254,"logger":"controller_v2alpha2activemqartemis","msg":"Updating ScalingState","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.4626012,"logger":"controller_v2alpha2activemqartemis","msg":"Updating status for pods","ActiveMQArtemis Name":"ex-aao"}
{"level":"info","ts":1591010205.462628,"logger":"controller_v2alpha2activemqartemis","msg":"Getting status for pods","ActiveMQArtemis Name":"ex-aao-ss"}
{"level":"info","ts":1591010205.4626887,"logger":"olm","msg":"Found deployments with status ","stopped":[],"starting":["ex-aao-ss-1"],"ready":[]}
{"level":"info","ts":1591010205.462715,"logger":"controller_v2alpha2activemqartemis","msg":"lastStatus.Ready len is \u0000"}
{"level":"info","ts":1591010205.462733,"logger":"controller_v2alpha2activemqartemis","msg":"status.Ready len is \u0000"}
`

It appears there is more work to do here on the reconciliation loop.